From 44b56e2f0973c1e1f4722f8d48464cf4b11d74e8 Mon Sep 17 00:00:00 2001
From: Bogdan Purcareata <bogdan.purcareata@nxp.com>
Date: Thu, 12 Oct 2017 12:55:41 +0000
Subject: [PATCH 126/767] staging: fsl-dpaa2/eth: Use signed variables for
 buffer accounting

When comparing each channel buf_count with the refill threshold and max
buffer per channel thershold, there's an implicit conversion to unsigned
which leads to underflow errors when per channel buffers are consumed
too fast.

Convert all variables to use signed types.

Signed-off-by: Bogdan Purcareata <bogdan.purcareata@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
index c50122dd8c13..27bd80aa4d9a 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
@@ -383,8 +383,8 @@ struct dpaa2_eth_priv {
 	u16 bpid;
 	u16 tx_qdid;
 	struct iommu_domain *iommu_domain;
-	u32 max_bufs_per_ch;
-	u32 refill_thresh;
+	int max_bufs_per_ch;
+	int refill_thresh;
 	bool has_xdp_prog;
 
 	void *cscn_mem;	/* Tx congestion notifications are written here */
-- 
2.17.0

